manifests/params.pp

```diff
@@ -112,17 +112,25 @@
     true    => pe_build_version(),
     default => undef
   }
-  # Not PE or pe_version < 2018.1.3, use PC1
-  if ($_pe_version == undef or versioncmp("${_pe_version}", '2018.1.3') < 0) {
+
+  # PE versions before 2018.1.3 use the Puppet 4 series of puppet agent
+  # releases. Outside of PE, we refer to this series of releases as 'PC1'. Note
+  # that although they include Puppet 4, the puppet-agent packages in this
+  # series are numbered '1.y.z' (later series _do_ match the version of puppet
+  # to that of the puppet-agent package, though).
+  if versioncmp("${::aio_agent_version}", '5.0.0') < 0 {
```

Comment:
Shouldn't this be based on the version we're upgrading to, not the one that's installed?

Reply:
The default behavior here is to install the most recent package that's compatible with the master's agent version, so it does make sense to check the existing version. I've added a comment about that.